File: p26/RemoveDuplicates.java
package leetcode.p26;

public class RemoveDuplicates {
    public static int removeDuplicates(int[] nums) {
        int j = 1;
        for (int i = 0; i < nums.length - 1; i++) {
            if (nums[i] != nums[i + 1]) {
                nums[j] = nums[i + 1];
                j++;
            }
        }
        return j;
    }

    public static void main(String[] args) {
        int[] nums = new int[]{0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
        int k = removeDuplicates(nums);
        for (int i = 0; i < k; i++) {
            System.out.println(nums[i]);
        }
    }
}

File: p0/p80/remove_duplicates.py
from typing import List


class Solution:
    """
    May 27, 2025 16:44
    Time Complexity: O(n)
    Space Complexity: O(1)
    """

    def removeDuplicates(self, nums: List[int]) -> int:
        prev = nums[0]
        index = 1
        count = 1
        for i in range(len(nums)):
            if nums[i] != prev:
                nums[index] = nums[i]
                index += 1
                prev = nums[i]
                count = 1
            else:
                if count < 2:
                    nums[index] = nums[i]
                    index += 1
                    count += 1
        return index
